ketcher-autotests/tests/Templates/Custom-Templates/custom-templates.spec.ts

```diff
@@ -0,0 +1,168 @@
+    await page.getByRole('tab', { name: 'Functional Groups' }).click();
+    await page.getByRole('tab', { name: 'Salts and Solvents' }).click();
+  });
+
+  test('Thumbnail images are readable in Template Dialog', async ({ page }) => {
+    /*
+   Test case: EPMLSOPKET-8932 
+   Open template dialog
+   Switch to "Functional Groups" tab
+   Observe some large structure
+   */
+    await selectFunctionalGroups(FunctionalGroups.Tf, page);
+    await clickInTheMiddleOfTheScreen(page);
+  });
+
+  test('Check automatically reset the search filter', async ({ page }) => {
+    /*
+   Test case: EPMLSOPKET-8906 
+    Enter the abbreviation in the search box (e.g. DMF)
+    Close 'Custom Templates' window
+    Open 'Custom Templates' window
+   */
+    await pressButton(page, STRUCTURE_LIBRARY_BUTTON_NAME);
+    await page.getByPlaceholder('Search by elements...').fill('DMF');
+    await page.getByPlaceholder('Search by elements...').press('Enter');
+    await takeEditorScreenshot(page);
+    await page.getByTestId('close-icon').click();
+    await pressButton(page, STRUCTURE_LIBRARY_BUTTON_NAME);
+  });
+
+  test('Adding template to canvas', async ({ page }) => {
+    /*
+   Test case: EPMLSOPKET-1667 
+   Click the 'Custom Templates' button.
+   Choose a template from any folder.
+   Click to add the selected template on the canvas.
+   */
+    await selectUserTemplatesAndPlaceInTheMiddle(
+      TemplateLibrary.Anthracene,
+      page,
+    );
+    await clickInTheMiddleOfTheScreen(page);
+  });
+
+  test('The scroll should not be displayed in the edit', async ({ page }) => {
+    /*
+   Test case: EPMLSOPKET-4736 
+   Launch Ketcher
+   Open Templates window
+   Open alpha-D-Sugars tab of Templates window
+   Click on Edit button in right down corner of template (e.g. alpha-D-Arabinofuranose)
+   */
+    await pressButton(page, STRUCTURE_LIBRARY_BUTTON_NAME);
+    await page.getByRole('tab', { name: 'Template Library' }).click();
+    await page.getByRole('button', { name: 'Aromatics (18)' }).click();
+    await page.getByTitle('Azulene').getByRole('button').click();
+    await takeEditorScreenshot(page);
+    await page.getByRole('button', { name: 'Edit' }).click();
+  });
+  test('Edit templates - Close window', async ({ page }) => {
+    /*
+   Test case: EPMLSOPKET-1700 
+   Click the 'Custom Template' button.
+   Click 'Edit' button for any template.
+   Click the 'X' button at the top right corner.
+   Click 'Edit' button for the same template.
+   Click the 'Cancel' button.
+   Click 'Edit' button for the same template.
+   Make any change(s) in the window. Click the 'Apply' button.
+   */
+    await pressButton(page, STRUCTURE_LIBRARY_BUTTON_NAME);
+    await page.getByRole('tab', { name: 'Template Library' }).click();
+    await page.getByRole('button', { name: 'Aromatics (18)' }).click();
+    await page.getByTitle('Azulene').getByRole('button').click();
+    await page.getByTestId('close-icon').click();
```

Comment:
3 lines of code are repeated here so it would look more accurate to use function like : 
async function someFunc() {
  await page.getByRole('tab', { name: 'Template Library' }).click();
  await page.getByRole('button', { name: 'Aromatics (18)' }).click();
  await page.getByTitle('Azulene').getByRole('button').click();
}

and then call it with a single line

Reply:
done